Tasks/Maven/maventask.ts

```diff
@@ -212,11 +223,55 @@ function createMavenSQRunner(sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?
 function applyPmdGoals(mvnRun: trm.ToolRunner):void {
     if (tl.getInput('pmdAnalysisEnabled', true) != 'true') {
         console.log('PMD analysis is not enabled');
+        return;
     }
     console.log('PMD analysis is enabled');
     pmd.applyPmdArgs(mvnb);
 }
 
+// Extract data from code analysis output files and upload results to build server
+function postCodeAnalysisResults():void {
```

Comment:
Discussed offline - I think that the reading of settings should stay in the main task so that pmdForMaven can be tested as an isolated unit. The remaining PMD-specific code is only lines 239-244.

Resolved, no action required.